### PR TITLE
Asset Server deplays queries until fully setup

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7231/app_12413_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7231/HighFidelity-Beta-PR12413-d6e281408144f0db5bb102ca4a0b99dd460cd63f.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7231/HighFidelity-Beta-PR12413-d6e281408144f0db5bb102ca4a0b99dd460cd63f.dmg

File: domain-server/src/AssetsBackupHandler.cpp [hunk 1]
```diff
@@ -9,52 +9,62 @@
...
-            // Give the Asset Server some time to bootup.
-            static constexpr int ASSET_SERVER_BOOTUP_MARGIN = 1 * 1000;
-            _mappingsRefreshTimer.start(ASSET_SERVER_BOOTUP_MARGIN);
+            // run immediately for the first time.
+            _mappingsRefreshTimer.start(0);
+        }
+    });
+    QObject::connect(nodeList.data(), &LimitedNodeList::nodeKilled, this, [this](SharedNodePointer node) {
+        if (node->getType() == NodeType::AssetServer) {
+            // run immediately for the first time.
```
Comment (huffman) [inline domain-server/src/AssetsBackupHandler.cpp]: comment copied from above?

File: domain-server/src/AssetsBackupHandler.cpp [hunk 2]
```diff
@@ -181,43 +197,65 @@ void BackupSupervisor::createBackup(QuaZip& zip) {
...
+        qCDebug(asset_backup) << "Could not find backup" << zip.getZipName() << "to restore.";
+
+        loadBackup(zip);
+
+        QuaZipDir zipDir { &zip, ZIP_ASSETS_FOLDER };
+
+        auto assetNames = zipDir.entryList(QDir::Files);
+        for (const auto& asset : assetNames) {
+            if (AssetUtils::isValidHash(asset)) {
+                if (!zip.setCurrentFile(MAPPINGS_FILE)) {
```
Comment (huffman) [inline domain-server/src/AssetsBackupHandler.cpp]: Should this be `asset` not `MAPPINGS_FILE`?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7242/app_12413_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7242/HighFidelity-Beta-PR12413-3297e39c144581788015b5ea4cd5bc52505fda0a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7242/HighFidelity-Beta-PR12413-3297e39c144581788015b5ea4cd5bc52505fda0a.exe

**Pull request merged**